sdk/core/azure-core/src/main/java/com/azure/core/util/Header.java

```diff
@@ -25,7 +33,37 @@
     public Header(String name, String value) {
         Objects.requireNonNull(name, "'name' cannot be null.");
         this.name = name;
-        this.value = value;
+        this.values = new LinkedList<>();
+        this.values.add(value);
+    }
+
+    /**
+     * Create a Header instance using the provided name and values.
+     *
+     * @param name the name of the header.
+     * @param values the values of the header.
+     * @throws NullPointerException if {@code name} is null.
+     */
+    public Header(String name, String... values) {
+        Objects.requireNonNull(name, "'name' cannot be null.");
+        this.name = name;
+        this.values = new LinkedList<>();
+        for (String value : values) {
+            this.values.add(value);
+        }
+    }
+
+    /**
+     * Create a Header instance using the provided name and values.
+     *
+     * @param name the name of the header.
+     * @param values the values of the header.
+     * @throws NullPointerException if {@code name} is null.
+     */
+    public Header(String name, List<String> values) {
+        Objects.requireNonNull(name, "'name' cannot be null.");
+        this.name = name;
+        this.values = values;
```

Comment:
Should we make a copy of the list instead of referencing the provided list to keep it immutable? This will also keep the `cachedStringValue` accurate.

Reply:
Yep, I've made it a copy operation now.